extensions/interactions/NumericInput/directives/numeric-input-validation.service.ts

```diff
@@ -163,6 +163,43 @@ export class NumericInputValidationService {
 
     return warningsList;
   }
+
+  getErrorString(value: string): string {
+    if (!value) {
+      return undefined;
+    }
+
+    value = value.trim();
+    const trailingDot = /\.\d/g;
+    const twoDecimals = /.*\..*\./g;
+    const extraChars = /[^0-9.+-]/g;
+    const trailingMinus = /^-/g;
+    const extraHyphen = /-.*-/g;
+
+    if (value.includes('.') && !value.match(trailingDot)) {
+      return 'Trailing decimals are not allowed.';
+    } else if (
+      value.match(twoDecimals)) {
+      return 'At most 1 decimal point should be present.';
+    } else if (
+      value.match(extraChars)) {
+      return 'Only use numbers, minus sign(-), and decimal(.)';
+    } else if (
+      value.includes('-') &&
+          !value.match(trailingMinus)) {
+      return 'Minus(-) sign is only allowed in beginning.';
+    } else if (
+      value.includes('-') && value.match(extraHyphen)) {
+      return 'At most 1 minus(-) sign should be present.';
+    }
```

Comment:
Should we return an empty string or `undefined` if there is no error?

Reply:
Empty string. Fixed.